dds/DCPS/InfoRepoDiscovery/DataWriterRemoteImpl.cpp

```diff
@@ -57,78 +45,43 @@ DataWriterRemoteImpl::add_association(const RepoId& yourId,
                std::string(reader_converter).c_str()));
   }
 
-  DataWriterCallbacks* parent = 0;
-  DDS::DataWriter_var dwv;
-  {
-    ACE_GUARD(ACE_Thread_Mutex, g, this->mutex_);
-    dwv = getDataWriter(this->parent_);
-    parent = this->parent_;
-  }
-  if (parent) {
-    parent->add_association(yourId, reader, active);
+  if (parent_.in()) {
+    parent_->add_association(yourId, reader, active);
```

Comment:
Isn't there a potential race condition when between the call to .in() and add_association() there would be a thread switch and another thread removes the parent?